src/main/java/frc/robot/subsystems/Elevator/ElevatorIOSim.java

```diff
@@ -0,0 +1,66 @@
+package frc.robot.subsystems.Elevator;
+
+import static edu.wpi.first.units.Units.MetersPerSecond;
+
+import edu.wpi.first.math.controller.ElevatorFeedforward;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.wpilibj.simulation.ElevatorSim;
+
+public class ElevatorIOSim implements ElevatorIO{
+
+    private ProfiledPIDController pidController;
+    private ElevatorFeedforward ffcontroller;
+
+    private double appliedVoltage;
+
+    private ElevatorSim elevatorSim;
+
+
+    public ElevatorIOSim() {
+        elevatorSim = new ElevatorSim(Constants.GEARBOX, Constants.GEARING,
+        Constants.CARRIAGE_MASS, Constants.DRUM_RADIUS,
+        Constants.MIN_HEIGHT, Constants.MAX_HEIGHT,
+        Constants.SIMULATE_GRAVITY, Constants.STARTING_HEIGHT);
+
+        pidController = new ProfiledPIDController(Constants.PROFILLED_PID_CONSTANTS.kP,
+        Constants.PROFILLED_PID_CONSTANTS.kI,
+        Constants.PROFILLED_PID_CONSTANTS.kD,
+        Constants.TRAPEZOID_PROFILE_CONSTRAINTS);
+        pidController.setTolerance(Constants.POSITION_TOLERANCE, Constants.VELOCITY_TOLERANCE);
+        pidController.setIZone(Constants.I_ZONE);
+
+        ffcontroller = new ElevatorFeedforward(Constants.FF_CONSTANTS.kS,
+        Constants.FF_CONSTANTS.kG,
+        Constants.FF_CONSTANTS.kV,
+        Constants.FF_CONSTANTS.kA);
+
+    }
+
+    public void updateInputs(ElevatorIOInputs inputs) {
+        elevatorSim.update(0.02); // adding this to global constants once this exists.
```

Comment:
go ahead and rename Constants.java (the global one) to global constants and put it in there